### ISSUE TITLE
Facing issue with probot getting error PEM routines:PEM_read_bio:no start line

### ISSUE THREAD

**Issue by hegdeashwin** [probot/friction#37]
Hello Probot Team,

**Ecosystem**

Node.js v10.0.0 or v8.11.2
npm v6.0.1 or above
Node.js app (probot app) deployed on `now.sh`
now.sh deployed URL [https://my-first-bot-dxujngefot.now.sh/probot](https://my-first-bot-dxujngefot.now.sh/probot)

**Problem**

- It seems like there is some like of authentication issue between `GitHub` and `now.sh`. Below I have added the logs which I am seeing in `now.sh`

- I have followed all kind of possible documents which I found out like [https://github.com/hoodiehq/first-timers-bot/pull/89](https://github.com/hoodiehq/first-timers-bot/pull/89), [https://medium.com/ahmed-t-ali/lets-build-a-github-pro-bot-5e155cec395f](https://medium.com/ahmed-t-ali/lets-build-a-github-pro-bot-5e155cec395f), [https://probot.github.io/docs/deployment/#now](https://probot.github.io/docs/deployment/#now) ... etc.

- I am also seeing my bot is correctly installed on my `hegdeashwin/guinea-pig` project. see screen-shots - see green right tickmark 

![screen shot 2018-06-01 at 9 39 53 pm](https://user-images.githubusercontent.com/4667864/40851289-97619344-65e4-11e8-80a5-81ee1cc62424.png)

![screen shot 2018-06-01 at 9 40 09 pm](https://user-images.githubusercontent.com/4667864/40851275-910422fa-65e4-11e8-8cf6-67e1e02a26f2.png)

**Expected**

- When I try to add issue or commit on exist issue, my bot should reply with some message

**Actual**

- By bot is not replying

**now.sh logs**

```
15:50:50.055Z ERROR event: error:0906D06C:PEM routines:PEM_read_bio:no start line (id=8e569370-65b3-11e8-8d0b-4af054e62370)
06/01 09:20 PM (9m)
  Error: error:0906D06C:PEM routines:PEM_read_bio:no start line
06/01 09:20 PM (9m)
      at Sign.sign (internal/crypto/sig.js:82:26)
06/01 09:20 PM (9m)
      at Object.sign (/home/nowuser/src/node_modules/jwa/index.js:76:45)
06/01 09:20 PM (9m)
      at Object.jwsSign [as sign] (/home/nowuser/src/node_modules/jws/lib/sign-stream.js:32:24)
06/01 09:20 PM (9m)
      at Object.module.exports [as sign] (/home/nowuser/src/node_modules/jsonwebtoken/sign.js:189:16)
06/01 09:20 PM (9m)
      at Robot.app (/home/nowuser/src/node_modules/probot/lib/github-app.js:12:16)
06/01 09:20 PM (9m)
      at Robot.auth (/home/nowuser/src/node_modules/probot/lib/robot.js:169:53)
06/01 09:20 PM (9m)
      at EventEmitter.events.on (/home/nowuser/src/node_modules/probot/lib/robot.js:107:33)
06/01 09:20 PM (9m)
      at Promise.all.handlers.map.handler (/home/nowuser/src/node_modules/promise-events/emitter.js:125:28)
06/01 09:20 PM (9m)
      at Array.map (<anonymous>)
06/01 09:20 PM (9m)
      at EventEmitter.emit (/home/nowuser/src/node_modules/promise-events/emitter.js:123:40)
06/01 09:20 PM (9m)
      at events.emit.then (/home/nowuser/src/node_modules/probot/lib/robot.js:30:26)
06/01 09:20 PM (9m)
      at process._tickCallback (internal/process/next_tick.js:68:7)
06/01 09:20 PM (9m)
  --
06/01 09:20 PM (9m)
  event: {
06/01 09:20 PM (9m)
    "id": "8e569370-65b3-11e8-8d0b-4af054e62370",
06/01 09:20 PM (9m)
    "event": "issues.opened",
06/01 09:20 PM (9m)
    "repository": "hegdeashwin/guinea-pig"
06/01 09:20 PM (9m)
  }
06/01 09:20 PM (9m)
15:50:50.055Z  INFO http: POST / 200 - 7.62 ms (id=8e569370-65b3-11e8-8d0b-4af054e62370)
06/01 09:20 PM (9m)
15:50:50.076Z  INFO http: POST / 200 - 19.53 ms (id=8e569370-65b3-11e8-8d0b-4af054e62370)
06/01 09:20 PM (9m)
15:50:50.078Z ERROR event: error:0906D06C:PEM routines:PEM_read_bio:no start line (id=8e569370-65b3-11e8-8d0b-4af054e62370)
06/01 09:20 PM (9m)
  Error: error:0906D06C:PEM routines:PEM_read_bio:no start line
06/01 09:20 PM (9m)
      at Sign.sign (internal/crypto/sig.js:82:26)
06/01 09:20 PM (9m)
      at Object.sign (/home/nowuser/src/node_modules/jwa/index.js:76:45)
06/01 09:20 PM (9m)
      at Object.jwsSign [as sign] (/home/nowuser/src/node_modules/jws/lib/sign-stream.js:32:24)
06/01 09:20 PM (9m)
      at Object.module.exports [as sign] (/home/nowuser/src/node_modules/jsonwebtoken/sign.js:189:16)
06/01 09:20 PM (9m)
      at Robot.app (/home/nowuser/src/node_modules/probot/lib/github-app.js:12:16)
06/01 09:20 PM (9m)
      at res.cache.wrap (/home/nowuser/src/node_modules/probot/lib/robot.js:162:55)
06/01 09:20 PM (9m)
      at process._tickCallback (internal/process/next_tick.js:68:7)
06/01 09:20 PM (9m)
  --
06/01 09:20 PM (9m)
  event: {
06/01 09:20 PM (9m)
    "id": "8e569370-65b3-11e8-8d0b-4af054e62370",
06/01 09:20 PM (9m)
    "event": "issues.opened",
06/01 09:20 PM (9m)
    "repository": "hegdeashwin/guinea-pig",
06/01 09:20 PM (9m)
    "installation": 197985
06/01 09:20 PM (9m)
  }
06/01 09:20 PM (9m)
15:50:59.856Z  INFO http: POST / 200 - 0.73 ms (id=9434f4d0-65b3-11e8-895e-a64757fbe9ad)
06/01 09:20 PM (9m)
15:50:59.870Z  INFO http: POST / 200 - 0.47 ms (id=9434f4d0-65b3-11e8-895e-a64757fbe9ad)
```

**Comment by abhijeetps** [comment]
Hi @hegdeashwin!
Thank you for opening this issue.
I explored your bot (https://github.com/hegdeashwin/my-first-bot/) and it appears that should work fine.
The problem behind the error that you are facing is with deploying your app on now.sh.

I assume that while deploying app on now.sh, you have followed all the steps mentioned here: https://probot.github.io/docs/deployment/#now

This error is usually thrown when you have entered wrong PEM file. Please recheck your PEM file and don't modify the PEM file.

Also, can you please share which OS are you using while deploying the app on now.sh?

You can also try Heroku and Glitch to deploy your app meanwhile. Most of the Probot apps are deployed on these two platforms.

Also, I searched a little and found this relatable: https://github.com/yagop/node-telegram-bot-api/issues/63

Hope it helps. 😄 

<blockquote><img src="https://avatars3.githubusercontent.com/u/4667864?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/hegdeashwin/my-first-bot">hegdeashwin/my-first-bot</a></strong></div><div>my-first-bot - My first bot & I love it ...</div></blockquote>
<blockquote><img src="https://probot.github.io/assets/card.png" width="48" align="right"><div><img src="/favicon.ico" height="14"> Probot</div><div><strong><a href="https://probot.github.io/docs/deployment/">Deployment</a></strong></div><div>GitHub Apps to automate and improve your workflow</div></blockquote>

**Comment by hegdeashwin** [comment]
@aps120797 Thanks for the quick reply. Let me try out Glitch and Netlify.
Closing this as of now, In case the issue still exists I may reopen it. Thanks once again

👍

**Comment by hegdeashwin** [comment]
@aps120797 and it very well worked!

The issue is with `now.sh` when we use `free plan without any custom or fix domain name` 😄 
`now.sh` generates new URL with a dynamic hash key in the URL at the time of deployment.

So, initially, I used the WebHook URL `my-first-bot.<A>.now.sh` and generated `private key pem` and then I deployed the Probot app which generated new URL `my-first-bot.<B>.now.sh`.

Thus, even if we change the WebHook URL in GitHub with `my-first-bot.<B>.now.sh` the `private key pem` which we generated does not match and throws an error.

In Heroku-CLI, 
first it gave me fix URL which I use it to generate my `private key pem` on GitHub and use it to deploy the Probot app later ... which solved the problem ... cool, Loved this.

<img width="793" alt="screen shot 2018-06-02 at 2 48 21 am" src="https://user-images.githubusercontent.com/4667864/40863958-72942ccc-660f-11e8-87bc-dafce15f4c36.png">
